compact_proposal/compact_proposal.cls

```diff
@@ -60,6 +61,45 @@
 \definecolor{lgrey}{gray}{0.75}		%	make a light grey color for mnotes
 
 
+%% Define note functions
+%%
+% make note for margin discussion.
+\NewDocumentCommand{\mnote}{o m}{%
+    \nottoggle{@nonotes}{%
+    % if not nonotes:
+        \iftoggle{@draft}{%
+        % if draft:
+            \IfValueTF{#1}%
```

Comment:
Some fo the stuff like this I don't really understand yet, but I think I see how the general structure of this works. You'll have to teach me the specifics when you get back.